application/common/models/Method.php

```diff
@@ -179,20 +188,64 @@ public static function createAndSendVerification($userId, $type, $value)
     }
 
     /**
+     * Checks for an existing unverified method and resends verification code if found
+     * @param integer $userId
+     * @param string $type
+     * @param string $value
+     * @return Method|null
+     */
+    public static function checkForExistingAndResend($userId, $type, $value)
+    {
+        $existing = Method::find()->where([
+            'user_id' => $userId,
+            'type' => $type,
+            'verified' => 0,
+        ])->andWhere([
+            '>=', 'verification_expires', Utils::getDatetime()
+        ])->all();
+
+        /** @var Method $existMethod */
+        foreach ($existing as $existMethod) {
+            if ($existMethod->type === self::TYPE_PHONE) {
+                if (Utils::stripNonNumbers($existMethod->value) === Utils::stripNonNumbers($value)) {
+                    $existMethod->sendVerification();
+                    return $existMethod;
+                }
+            } else {
+                if (strtolower($existMethod->value) === strtolower($value)) {
+                    $existMethod->sendVerification();
+                    return $existMethod;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
      * Send verification to either email or phone based on $this->type
      * @throws \Exception
      */
     public function sendVerification()
     {
+        /*
+         * Count as verification attempt and send verification code
+         */
+        $this->verification_attempts++;
+
         if ($this->type == self::TYPE_EMAIL) {
             $this->sendVerificationEmail();
         } elseif ($this->type == self::TYPE_PHONE) {
             $this->verification_code = $this->sendVerificationPhone();
-            if ( ! $this->save()) {
-                throw new \Exception('Unable to save method after sending phone verification', 1461441850);
-            }
         } else {
-            throw new \Exception('Invalid method type', 1461432437);
+            throw new BadRequestHttpException('Invalid method type', 1461432437);
+        }
+
+        if ( ! $this->save()) {
```

Comment:
As we did somewhere else, should we make sure we successfully save the updated attempts-counter before sending the verification message (to avoid situations where a failure to save the updated attempts-counter could let someone send an unlimited number of verification messages)?